http-server/src/server.rs

```diff
@@ -344,6 +344,26 @@ impl<M> Builder<M> {
 			health_api: self.health_api,
 		})
 	}
+
+	/// Returns a service that can be utilised with `tower` compatible crates.
+	pub fn to_service(self, methods: impl Into<Methods>) -> Result<TowerService<M>, Error> {
```

Comment:
it might be better to implement this on the `Server struct` and not the builder.

Then you don't need the `Option` as `remote addr`

Reply:
Have managed to tidy up the API, without requiring us to have `Options` on either the service or server, let me know what you think of it 😄

Reply:
looks really good